pyquil/api/_devices.py

```diff
@@ -16,35 +16,66 @@
 from pyquil.api._base_connection import get_session, get_json
 from pyquil.api._config import PyquilConfig
 from pyquil.device import Device
+from pyquil.api._errors import UnknownApiError
 
 
 def list_devices():
     """
-    Query the Forest 2.0 server for its knowledge of QPUs.
+    Query the Forest 2.0 server for a list of underlying QPU devices.
 
-    :return: A dictionary, keyed on device names. Each value is a dictionary of the form
-    {
-        "is_online":   a boolean indicating the availability of the device,
-        "is_retuning": a boolean indicating whether the device is busy retuning,
-        "specs":       a Specs object describing the entire device, serialized as a dictionary,
-        "isa":         an ISA object describing the entire device, serialized as a dictionary,
-        "noise_model": a NoiseModel object describing the entire device, serialized as a dictionary
-    }
+    NOTE: These can't directly be used to manufacture pyQuil Device objects, but this gives a list
+          of legal values that can be supplied to list_lattices to filter its (potentially very
+          noisy) output.
+
+    :return: A list of device names.
     """
+    # For the record, the dictionary stored in "devices" that we're getting back is keyed on device
+    # names and has this structure in its values:
+    #
+    # {
+    #   "is_online":   a boolean indicating the availability of the device,
+    #   "is_retuning": a boolean indicating whether the device is busy retuning,
+    #   "specs":       a Specs object describing the entire device, serialized as a dictionary,
+    #   "isa":         an ISA object describing the entire device, serialized as a dictionary,
+    #   "noise_model": a NoiseModel object describing the entire device, serialized as a dictionary
+    # }
+
     session = get_session()
     config = PyquilConfig()
 
-    return get_json(session, config.forest_url + "/devices")["devices"]
+    return list(get_json(session, config.forest_url + "/devices")["devices"].keys())
```

Comment:
consider replacing `list(` with `sorted(`, which will still turn an iterator into a list. I do this as force-of-habit to make our world just a little bit more deterministic

Reply:
Downsides of having me as a cocontributor: I don't know much python.